enumerables.rb

```diff
@@ -0,0 +1,216 @@
+module Enumerable
+  def my_each
+    return unless is_a? Array
+
+    return to_enum unless block_given?
+
+    i = 0
+    while i <= length - 1
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    object_to_return = []
+    if is_a? Range
+      to_a.my_each { |elt| object_to_return.push(elt) if yield(elt) }
+      return object_to_return
+    end
+    my_each do |element|
+      object_to_return << element if yield(element)
+    end
+    object_to_return
+  end
+
+  def my_all?(*arg, &block)
+    if arg.length == 1
+      i = 0
+      if arg[0].is_a? Class
+        while i < length
+          return false unless self[i].is_a? arg[0]
+
+          i += 1
+        end
+        return true
+      else
+        while i < length
+          return false if self[i] != arg
+
+          i += 1
+        end
+      end
+      return true
+    end
+    block = lambda { |obj| obj } unless block_given?
+
+    i = 0
+    while i < length
+      return false unless block.call(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?(*arg, &block)
+    if arg.length == 1
+      i = 0
+      if arg[0].is_a? Class
+        while i < length
+          return true if self[i].is_a? arg[0]
+
+          i += 1
+        end
+        false
+      else
+        i = 0
+        while i < length
+          return true if self[i] == arg
+
+          i += 1
+        end
+      end
+      false
+    end
+    block = lambda { |obj| obj } unless block_given?
+
+    i = 0
+    while i < length
+      return true if block.call(self[i])
+
+      i += 1
+    end
+    false
+  end
```

Comment:
## Class integer
- Passing a class integer as parameter.
 - [ ] Please fix this issue 
Example:
```ruby
p %w[cat mouse].any?(Integer)
# => expected: false
p %w[cat mouse].my_any?(Integer)
# => got: true
```